test/powershell/Language/Scripting/HashtableToPSCustomObjectConversion.Tests.ps1

```diff
@@ -0,0 +1,248 @@
+﻿Describe "Tests for hashtable to PSCustomObject conversion" -Tags "CI" {
+
+    Context 'New-Object cmdlet should accept empty hashtable or $null as Property argument' {
+            
+        { $script:a = new-object psobject -property $null } | Should not throw
+        
+        It '$a should not be $null' { $script:a | Should Not Be $null }
+        It '$a type' { $script:a | should BeOfType "System.Management.automation.psobject" }
+    }
+
+    It 'New-Object cmdlet should throw terminating errors when user specifies a non-existent property or tries to assign incompatible values' {
+       
+        try
+        { 
+            $source = @"
+                        public class SampleClass6
+                        {
+                            public SampleClass6(int x)
+                            {
+                            	a = x;
+                            }
+                            
+                            public SampleClass6()
+                            {
+                            }
+                                                          
+                            public int a;
+                            public int b;
+                        }
+"@
+            add-type -typedefinition $source
+            New-Object -TypeName SampleClass6 -Property @{aa=10;b=5}
+            Throw "Exception expected, execution should not have reached here"            
+        }
+        catch {
+            $_.Exception | Should BeOfType "InvalidOperationException"
+            $_.Exception.Message | Should Match "aa"
+            $_.FullyQualifiedErrorId | should be "InvalidOperationException,Microsoft.PowerShell.Commands.NewObjectCommand"           
+        }
+    }

```

Comment:
I've been thinking about this one, I _think_ it's just as valid to use System.Management.Automation.Host.Coordinates as follows:
```
   It 'New-Object cmdlet should throw terminating errors when user specifies a non-existent property or tries to assign incompatible values' {
        try {
            New-Object -TypeName System.MAnagement.Automation.Host.Coordinates -Property @{xx=10;y=5}
            Throw "Exception expected, execution should not have reached here"            
        }
        catch {
            $_.Exception | Should BeOfType "InvalidOperationException"
            $_.Exception.Message | Should Match "xx"
            $_.FullyQualifiedErrorId | should be "InvalidOperationException,Microsoft.PowerShell.Commands.NewObjectCommand"           
        }
    }
```
that way we can avoid a call to new-object, which would be good!